### DISCUSSION TITLE
HA errore: "Sistema non integro - Non privilegiato"

### DISCUSSION THREAD

**Discussion by MapoCaos** [indomus/forum#2947] [HUB: Home Assistant]
Ciao a tutti, quest'inverno ho installato HA su un mini pc magician (recensito su indomus) e acquistato apposta per l'uso di AM (ed in futuro pi-hole). Ho eseguito l'installazione supervised perchè non conosco linux ma volevo poter installare pi-hole in seguito. Dopo relativamente poco tempo (non so se a causa di un aggiornamento) mi è apparso questo avviso: "Il sistema non è attualmente integro perché non ha accesso privilegiato al runtime docker. Usa il link per saperne di più e come risolvere questo problema" il link mi parla solo di HAOS ed è poco utilie (https://www.home-assistant.io/more-info/unhealthy/privileged). Dato che avevo fatto delle porve su linux pensavo di aver fatto casino cosi da niubbo, ho reinstallato Debian 12, docker e supervide e HA, come da guida indomus. Poi avevo effettuato backup giornalieri su google drive, cosi ho scaricato il backup e l'ho messo su HA. Tutto è andato bene però dopo il primo aggiornamento (che ci ha messo parecchio tempo) e ritornato l'avviso "critico" datato 5/01/24. A questo punto credo che il problema sia nel bakcup e devo azzerare tutto anche li o riparare qualcosa (non ho idea di come fare ne a disinstallare ne pulire ne a ripararlo, so solo reinstallre da capo Deb 12 e tutto il resto). Online non ho trovato molto su questo errore. 
A parte l'avviso avevo notato che il AH ha alcuni problemi che non so se siano dovuto a questo ma credo/spero di si. Aggiornamenti sia di HA che di componenti aggiuntivi lentissimi (anche di un ora o più) Altro problema con un ups APC, sono riuscito a collegarlo ad HA  ma creando un automazione che quando la tensione di ingresso si riduce mi deve mandare una notifica e un messaggio su google home (giusto per provare se funzionava) ha funzionato in poche prove e soprattutto con minuti in ritardo (nonostante il grafico su HA della tensione di ingresso mostrasse un azzeramento). Io credo che questo comportamento sia dovuto all'errore perchè per come gira adesso è inutilizzabile ed inutile. Qualcuno ha qualche consiglio? grazie in anticipo a tutti

**Comment by snakuzzo** [comment]
> Ho eseguito l'installazione supervised perchè non conosco linux 

E' bastato leggere questo per capire che hai completamente toppato il tipo di installazione.

Come abbiamo ripetuto decine di volte, e come viene esplicitamente indicato nella [documentazione ufficiale](https://github.com/home-assistant/supervised-installer), questo tipo di installazione è per utenti avanzati.

HA Supervised noi la sconsigliamo a prescindere. Questo perché, è vero che avrai sostanzialmente HA OS e poi tutta la macchina libera per fare altre cose, ma di fatto non sei così libero come credi. Ogni piccola dipendenza/aggiornamento a livello di sistema operativo può rendere "unhealty" l'installazione di HA.
Per cui il consiglio è... fai un backup completo, butta nel secchio questa installazione e, se non hai intenzione di usare Core su Docker ma vuoi gli addon, il supervisor e tutto il cucuzzaro, valuta queste due opzioni:

- Installa un virtualizzatore (es. VirtualBox) e crea una VM con l'immagine di HA OS (personalmente ti consiglio questa)
- Non installare Debian e installa direttamente HA OS

Una volta installato, carichi il backup fatto prima e hai finito.

P.S. alcuni/molti usano Proxmox e poi creano VM di HA OS. Si può fare, ma noi non trattiamo Proxmox e non riceverai alcun consiglio su come usare Proxmox

**Reply by MapoCaos** [reply]
Grazie mille per la risposta, mi scuso per la mia risposta in ritardo ma non ho potuto fare altrimenti. 
Ok, purtroppo attualmente non ho tempo per imparare ad usare core su docker (magari in futuro) quindi credo che andrò con la tua prima opzione: virtualbox con HA OS. Io voglio attenermi ai casi consigliati (dato che nn sono io l'esperto) e in cui possibilmente posso chiedervi consigli (quindi Proxmox è escluso a prescindere). 
Quindi ora ho due domande:
1) Come faccio a creare una macchina virtuale con HA OS? su indomus non mi sembra di aver trovato guide oltre alle tre opzioni: HA OS, Supervised, Core. Non ho cercato abbastanza bene? 

2) A questo punto voglio disinstallare supervised e HA e docker dato che non mi servono più, ma non so come fare? Avevo già cercato e provato prima di reinstallare direttamente Debian ma non sono mai riuscito. Avete qualche suggerimento?

Grazie mille

**Reply by snakuzzo** [reply]
1. Non abbiamo una guida...ma ti basta cercare su Google e troverai tutte le risposte su come creare una VM

2. Innanzi tutto, prima di procedere, fai un backup completo di HA. Poi...Supervised di fatto non è altro che un insieme di container Docker. Quindi devi comunque conoscere alcuni comandi docker. Diciamo che dovrebbe bastarti questo comando... altrimenti vai di Google e cerca come fermare e  rimuovere tutti i container

```
docker ps -aq | xargs docker stop | xargs docker rm
```

**Reply by MapoCaos** [reply]
Ok grazie mille! provo a fare come mi hai consigliato

Grazie ancora

**Comment by MapoCaos** [comment]
Sono riuscito ad installare virtualbox con HA OS, l'unica cosa che proprio non riesco a impostare è che la macchina virtuale con HA OS si avvii al reboot del sistema operativo (debian 12) essendoci tanti tipi di linux non ho trovato niente per debian 12 con interfaccia GNOME (ma neanche tramite terminale). Ho installato Tweaks ma mi permette di aprire Virtualbox, però non la macchina virtuale. Io sono riuscito a creare una scorciaotia su desktop e ho copiato il percorso della VM però su Tweaks non mi permette di copiare il percorso che voglio che apra. Poi mi sembra che Tweaks apra virtualbox solo una volta effettuato il login come utente, cosa che comunque non sarebbe l'impostazione corretta. Per caso hai dei consigli? grazie

**Reply by indomus** [reply]
domanda cretina: ma se sei in grado di utilizzare debian, qual è il senso di installare Home Assistant OS su macchina virtuale? 🤷‍♂️

**Reply by snakuzzo** [reply]
@MapoCaos prova...
https://forums.virtualbox.org/viewtopic.php?t=93997

**Reply by MapoCaos** [reply]
Grazie @snakuzzo! provo, ho visto che li linkano il manuale per l'avvio della VM durante il boot del sistema.

Rispondo "brevemente" a @indomus. Io vorrei utilizzare un pc su cui far girare HA, ma non dedicarlo esclusivamente a quello perchè in futuro vorrei aver la possibilità di (almeno provare) installare pi-hole (al momento uso nextdns). Quindi ho escluso di l'installazione di HA OS. In realtà non so usare debian molto bene ma con le guide e forum cerco di raccapezzarmi e al momento non ho troppo tempo per imparare bene, ne debian ne HA Core, cosi avevo escluso di usare core e avevo optato per l'installazione supervised unendo cosi le le caratteristiche delle altre due installazioni. 
Però Snakuzzo mi ha fatto notare che la supervised non è consigliata per i principianti e quindi in base alle mi necessità (avere HA "completo" con tutto il mappazzone e avere la possibilità di far girare altri programmi su Debian) mi ha consigliato di far girare HA su una macchina virtuale. Così sto cambiando il tipo di installazione.

**Reply by indomus** [reply]
ma infatti il consiglio nascosto era ed è quello di usare home assistant core@docker. home assistant os non lo consigliamo quasi mai, men che meno su macchina virtuale, supervised poi è una cosa di mezzo (forse la peggiore delle scelte).

**Reply by MapoCaos** [reply]
Proverò ancora a far partire virtualbox, ma se avrò poca fortuna farò che seguire il consiglio e installerò direttamente core, mi sa che alla fine è l'unica..
Grazie mille!

**Comment by parente89** [comment]
> Ciao a tutti, quest'inverno ho installato HA su un mini pc magician (recensito su indomus) e acquistato apposta per l'uso di AM (ed in futuro pi-hole). Ho eseguito l'installazione supervised perchè non conosco linux ma volevo poter installare pi-hole in seguito. Dopo relativamente poco tempo (non so se a causa di un aggiornamento) mi è apparso questo avviso: "Il sistema non è attualmente integro perché non ha accesso privilegiato al runtime docker. Usa il link per saperne di più e come risolvere questo problema" il link mi parla solo di HAOS ed è poco utilie (https://www.home-assistant.io/more-info/unhealthy/privileged). Dato che avevo fatto delle porve su linux pensavo di aver fatto casino cosi da niubbo, ho reinstallato Debian 12, docker e supervide e HA, come da guida indomus. Poi avevo effettuato backup giornalieri su google drive, cosi ho scaricato il backup e l'ho messo su HA. Tutto è andato bene però dopo il primo aggiornamento (che ci ha messo parecchio tempo) e ritornato l'avviso "critico" datato 5/01/24. A questo punto credo che il problema sia nel bakcup e devo azzerare tutto anche li o riparare qualcosa (non ho idea di come fare ne a disinstallare ne pulire ne a ripararlo, so solo reinstallre da capo Deb 12 e tutto il resto). Online non ho trovato molto su questo errore. A parte l'avviso avevo notato che il AH ha alcuni problemi che non so se siano dovuto a questo ma credo/spero di si. Aggiornamenti sia di HA che di componenti aggiuntivi lentissimi (anche di un ora o più) Altro problema con un ups APC, sono riuscito a collegarlo ad HA ma creando un automazione che quando la tensione di ingresso si riduce mi deve mandare una notifica e un messaggio su google home (giusto per provare se funzionava) ha funzionato in poche prove e soprattutto con minuti in ritardo (nonostante il grafico su HA della tensione di ingresso mostrasse un azzeramento). Io credo che questo comportamento sia dovuto all'errore perchè per come gira adesso è inutilizzabile ed inutile. Qualcuno ha qualche consiglio? grazie in anticipo a tutti

**Per risolvere questo problema, puoi seguire questi passaggi:**

Assicurati che l'utente che esegue Home Assistant abbia i permessi necessari per accedere a Docker. Potresti dover aggiungere l'utente al gruppo Docker se non lo hai già fatto. Puoi farlo con il seguente comando da terminale:

**sudo usermod -aG docker <nome_utente>**

Dopo aver aggiunto l'utente al gruppo Docker, riavvia il servizio Docker per applicare le modifiche. Puoi farlo con il seguente comando:

**sudo systemctl restart docker**

Riavvia Home Assistant per vedere se l'errore è stato risolto.
